project/Settings.scala

```diff
@@ -116,7 +119,8 @@ object Settings {
     crossScalaVersions := supportedScalaVersions,
 
     scalacOptions ++= Seq(
-      "-Xsource:2.12"
+      "-Xsource:2.12",
+      "-Xmax-classfile-name", "240"
```

Comment:
We need to explicitly set this value in order to avoid this [issue](https://stackoverflow.com/questions/45022231/getting-file-name-too-long-when-running-tests-on-circleci).

As of Scala 2.13, the default value for this setting is 240. 

See: 
- https://github.com/scala/bug/issues/3623
- https://github.com/scala/scala/pull/7497

Reply:
can we move scala 2.13 config stuff to a different PR ?

Reply:
Jus to clarify, below setting is needed for 2.11 and 2.12 versions.

```
"-Xmax-classfile-name", "240"
```

For 2.11 and 2.12 the default value for that setting is `255`. `255` casues the issue mentioned [here](https://stackoverflow.com/questions/45022231/getting-file-name-too-long-when-running-tests-on-circleci). As of Scala 2.13 the default value for that setting is `240` (they fixed the issue by changing the default value to a proper value for you), which essentially means that if our main Scala version was 2.13, the setting wouldn't be necessary, but is is for now! :)